src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -312,14 +302,16 @@ public PublicKey PublicKey
                 ThrowIfInvalid();
 
                 PublicKey? publicKey = _lazyPublicKey;
+
                 if (publicKey == null)
                 {
                     string keyAlgorithmOid = GetKeyAlgorithm();
-                    byte[] parameters = GetKeyAlgorithmParameters();
-                    byte[] keyValue = GetPublicKey();
+                    byte[] parameters = Pal.KeyAlgorithmParameters;
```

Comment:
The public APIs `GetKeyAlgorithmParameters` and `GetPublicKey` create defensive copies, then we were feeding them in to `AsnEncodedData`, which again created defensive copies. If we go to the PAL directly for the bytes we can avoid the first defensive copy.